### PR TITLE
Update plugin configuration cache compatibility table

### PR DESCRIPTION
**Pull request by mlopatkin**

### CODE REVIEW

**Comment by mlopatkin** [comment]
[configuration_cache.html](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/59249632:id/distributions/gradle-8.0-docs.zip!/gradle-8.0-20221208165532%2B0000/docs/userguide/configuration_cache.html#config_cache:plugins)

**Comment by mlopatkin** [comment]
I wonder if we should also mention the change of status of these plugins in the release notes. Even though no plugins besides maybe build-init actually became compatible in 8.0, I don't think we announced some of the support status transitions before.

**Comment by bamboo** [comment]
Good point, @mlopatkin. We should announce when plugins become compatible with the configuration cache in the release notes 👍

**Review by bamboo — COMMENTED** [review]
🎉

**Comment by mlopatkin** [comment]
@bot-gradle test BD

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -224,6 +224,14 @@ There are some additional advantages to this new behaviour:
 
 This consistent behavior for cache miss and cache hit builds should help people who are migrating to use the configuration cache, as many more problems can now be discovered on the first (cache miss) build.
 
+#### Core plugins compatibility updates
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:227]:
```suggestion
#### Improved compatibility with core plugins 
```

Comment (hythloda) [reply subprojects/docs/src/docs/release/notes.md:227]: Reworded so that the title is consistent with others

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**